### DISCUSSION TITLE
Queues with sharding

### DISCUSSION THREAD

**Discussion by lumio-aamadeo** [taskforcesh/bullmq#1326] [Ideas]
Is there any plan to add the ability to a queue to be sharded? For example I may want throughput so I use a lot of workers, but I dont want to jobs related to a user to execute concurrently, I could solve that by having queues with concurrency factor of 1 and have 1 queue per shard, but that would involve maintaining a behavior that probably should be handled natively in this library.

**Comment by manast** [comment]
Maybe groups with concurrency max 1 does what you need: https://docs.bullmq.io/bullmq-pro/groups/concurrencyhttps://docs.bullmq.io/bullmq-pro/groups/concurrency
